crates/api_crud/src/comment/create.rs

```diff
@@ -164,6 +173,15 @@ pub async fn create_comment(
   )
   .await?;
 
+  // Update the read comments, so your own new comment doesn't appear as a +1 unread
+  update_read_comments(
+    local_user_view.person.id,
+    post_id,
+    post_view.counts.comments + 1,
+    &mut context.pool(),
+  )
+  .await?;
```

Comment:
We should do a db transaction here for all these queries.

Reply:
That should be done by moving things to triggers

Reply:
There's quite a lot here, so I'd prefer transactions be done in a different PR.

Side point, but it'd be nice if we could have all the functions in DB_schema take in a type that could be either the pool, or an individual connection, so that we could support transactions better. I'll make an issue for that.

Reply:
Most functions already use this enum:

https://github.com/LemmyNet/lemmy/blob/ec77c00ef80248e9eceec800082e34547f9aa084/crates/db_schema/src/utils.rs#L68

Reply:
Oh nice, I didn't realize DbPool was already an enum, I thought it was only the pool. Should probably rename that.